react/features/chat/components/AbstractMessageContainer.js

```diff
@@ -32,15 +37,25 @@ export default class AbstractMessageContainer<P: Props> extends PureComponent<P>
         const groups = [];
         let currentGrouping = [];
         let currentGroupParticipantId;
+        const localParticipant = getLocalParticipant(APP.store.getState());
 
         for (let i = 0; i < messagesCount; i++) {
             const message = this.props.messages[i];
+            const splitedMessage = message.message.split('_');
+            const akAddress = splitedMessage.length > 1 ? splitedMessage.slice(0, 2).join('_') : '';
+            const text = splitedMessage.pop();
+
+            message.akAddress = akAddress;
+            message.message = text;
+
+            if (localParticipant.name === message.displayName && message.akAddress) {
+                message.messageType = 'local';
+            }
 
             if (message.id === currentGroupParticipantId) {
                 currentGrouping.push(message);
             } else {
                 currentGrouping.length && groups.push(currentGrouping);
-
                 currentGrouping = [ message ];
                 currentGroupParticipantId = message.id;
             }
```

Comment:
1) when we reload page, our local messages firstly display like remote.
2) tip button disappear, when you send 2 or more messages in a row, because we lose ak_address.